evaluator/src/main/resources/rules/osadl.rules.kts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// Use this rule like:
+//
+// $ ort evaluate -i scanner/src/funTest/assets/dummy-expected-output-for-analyzer-result.yml --rules-resource /rules/osadl.rules.kts
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.decodeFromStream
+
+enum class Compatibility {
+    INHERENT,
+    YES,
+    NO,
+    UNKNOWN;
+
+    companion object {
+        fun fromString(value: String): Compatibility =
+            when (value) {
+                "Same" -> INHERENT
+                "Yes" -> YES
+                "No" -> NO
+                "Unknown", "Check dependency" -> UNKNOWN
+                else -> throw IllegalArgumentException("Unhandled compatibility '$value' found.")
+            }
+    }
+}
+
+@ExperimentalSerializationApi
+val matrix = javaClass.getResourceAsStream("/rules/osadl-matrix.json").use {
+    Json.Default.decodeFromStream<Map<String, Map<String, String>>>(it)
+}.mapValues { (_, outbounds) -> outbounds.mapValues { (_, value) -> Compatibility.fromString(value) } }
+
+fun PackageRule.LicenseRule.isOsadlMatrixCompliant(outbound: SpdxSingleLicenseExpression) =
+    object : RuleMatcher {
+        private val compatibleValues = enumSetOf(Compatibility.INHERENT, Compatibility.YES)
+
+        override val description = "isOsadlMatrixCompliant($license)"
+
+        @ExperimentalSerializationApi
+        override fun matches(): Boolean {
+            // Be conservative and use the simple license string without the exception string for lookup. Eventually
+            // resulting false-positives are better than an incompatibility to be missed.
+            val inbound = license.simpleLicense()
+            return matrix.getValue(inbound).getValue(outbound.simpleLicense()) in compatibleValues
+        }
+    }
+
+val ruleSet = ruleSet(ortResult, licenseInfoResolver) {
+    val licenseView = LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED
+
+    // Define a rule that is executed for each dependency.
```

Comment:
Could you add a comment explaining why the licenses of the dependencies only need to be checked against the project licenses. E.g to me it is unclear why the matrix is checked like this: e.g. why don't the licenses of the dependencies need to be compatile with each other, or at least with their ancestors in the dependency tree.

Reply:
> E.g to me it is unclear why the matrix is checked like this: e.g. why don't the licenses of the dependencies need to be compatile with each other, or at least with their ancestors in the dependency tree.

@LeChasseur, could you chime in here and repeat what you told me during our meeting about that?

Reply:
It depends on the purpose of the compatibility check:
1.
If I want to check compatibility with the project license, it is sufficient to check compatibility of the licenses of the dependencies (one-way) with the project license.
2.
If I want to check if the involved licenses are compatible in general (independent of the outbound license), it is necessary to check the compatibility of all licenses among each other. One use case is to verify if I can combine code under several licenses in one new project.

Reply:
Thanks @LeChasseur  @sschuberth !